.github/workflows/lint.yml

```diff
@@ -1,21 +1,102 @@
-# Runs linting related jobs.
+# Runs linting related jobs (local copy with LLVM for clippy/doc).
 
 name: lint
 
-# Limits workflow concurrency to only the latest commit in the PR.
-concurrency:
-  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
-  cancel-in-progress: true
-
 on:
   push:
-    branches: [main, next]
+    branches: [ main, next ]
   pull_request:
-    types: [opened, reopened, synchronize]
-
-permissions:
-  contents: read
+    types: [ opened, reopened, synchronize ]
 
 jobs:
-  lint:
-    uses: 0xMiden/.github/.github/workflows/lint.yml@main
+  typos:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: typos
+      - run: make typos-check
+
+  rustfmt:
+    name: rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Rustup
+        run: |
+          rustup update --no-self-update nightly
+          rustup +nightly component add rustfmt
+      - uses: Swatinem/rust-cache@v2
+        with:
+          save-if: ${{ github.event_name == 'push' && github.ref == format('refs/heads/{0}', inputs.target_branch) }}
+      - name: Fmt
+        run: make format-check
+
+  clippy:
+    name: clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Added: LLVM/Clang for RocksDB/bindgen
+      - name: Install LLVM/Clang
+        uses: KyleMayes/install-llvm-action@v2
```

Comment:
It is a bit unfortunate that we have to revert using "standard" lints file - but I'm not seeing a better way to do this.

cc @mmagician and @huitseeker in case you have some suggestions.